tableaudocumentapi/containerfile.py

```diff
@@ -16,19 +16,21 @@ def temporary_directory(*args, **kwargs):
         shutil.rmtree(d)
 
 
-def find_file_in_zip(zip, ext):
+def find_file_in_zip(zip):
     for filename in zip.namelist():
-        if os.path.splitext(filename)[-1].lower() == ext[:-1]:
+        try:
+            ET.parse(zip.open(filename)).getroot().tag in (
+                'workbook', 'datasource')
             return filename
+        except ET.ParseError:
+            # That's not an XML file by gosh
+            pass
 
 
 def get_xml_from_archive(filename):
-    file_type = os.path.splitext(filename)[-1].lower()
-    with temporary_directory() as temp:
-        with zipfile.ZipFile(filename) as zf:
-            zf.extractall(temp)
-            xml_file = find_file_in_zip(zf, file_type)
-            xml_tree = ET.parse(os.path.join(temp, xml_file))
+    with zipfile.ZipFile(filename) as zf:
+        xml_file = zf.open(find_file_in_zip(zf))
```

Comment:
Is this a context object also?  Or does it automatically get closed when the zipfile is closed?